src/main/scala/com/wavesplatform/database/LevelDBWriter.scala

```diff
@@ -710,20 +792,72 @@ class LevelDBWriter(writableDB: DB, fs: FunctionalitySettings, val maxCacheSize:
     readOnly(db => db.get(Keys.heightOf(blockId)).map(h => db.get(Keys.score(h))))
   }
 
-  override def loadBlockHeaderAndSize(height: Int): Option[(BlockHeader, Int)] = {
-    readOnly(_.get(Keys.blockHeader(height)))
+  override def loadBlockHeaderAndSize(height: Int): Option[(BlockHeader, Int)] = readOnly { db =>
+    db.get(Keys.blockHeaderAndSizeAt(Height(height)))
   }
 
-  override def loadBlockHeaderAndSize(blockId: ByteStr): Option[(BlockHeader, Int)] = {
-    readOnly(db => db.get(Keys.heightOf(blockId)).flatMap(h => db.get(Keys.blockHeader(h))))
+  override def loadBlockHeaderAndSize(blockId: ByteStr): Option[(BlockHeader, Int)] = readOnly { db =>
```

Comment:
There's no point in creating a read-only DB instance when retrieving just one value from.